src/tequila/objective/objective.py

```diff
@@ -1322,6 +1323,9 @@ class Variable:
     def name(self):
         return self._name
 
+    def __lt__(self, other):
```

Comment:
Will break functionality

Variables can be any hashable type
e.g. you can have one variable "a" and another (1,2,3) and a third (1,2,"a")
"a"<(1,2,3) will throw an assertion

Reply:
you could compare the hashes themselves: hash(name1) < hash(name2, that should work for all possible combinations

Reply:
So just be clear: are you saying that a tequila Variable can have another tequila Variable inside? As in, "a" is a variable and (1, 2, "a") is another variable which also somehow contains the first variable "a"? If so then yeah, I'll use .sort(key=myfunc) and sort by hash(var.name), but if not, then I don't see why comparing by str(var.name) is a problem. If anything it will be more intuitive for future developers to follow. I'm not trying to transform the tequila Variable into a string, just sort them by their names as strings. So str("a")<str((1, 2, 3)) will not throw an assertion, and extract_variables still returns a list of Variables, just sorted by their names if they were cast to string.

Reply:
Okay, so then can I put \_\_lt\_\_() with _return hash(self.name) < hash(other.name)_ ?

Reply:
Yes, that should work